### DISCUSSION TITLE
Router package

### DISCUSSION THREAD

**Discussion by artalar** [reatom/reatom#787] [Ideas]
We need a router package for the web. Simple, type-safe, scalable (it easy with Reatom). I think the best base, which we could fork and reuse is https://github.com/nanostores/router.

What we need:

- fork and rewrite nanostores/router to Reatom
- add and sync https://www.reatom.dev/packages/undo
- Add `block` API (https://github.com/remix-run/history/blob/dev/docs/blocking-transitions.md)
- Add `transition` API - the block for a particular route (for prefetching / animations)
- Improve API - each route should includes all needed APIs: `go`, `isCurrent`...?

Let me know if you could handle even the first task, it would be helpful.

Another resources.

- https://github.com/lukeed/regexparam

**Comment by vostrik** [comment]
@artalar hi there! 

I want to help with the first task at least. If it will be ok we can move further together. It is very interesting for me.

NB: I am familiar with Reatom only as spectator and reader.

Can I get `fork and rewrite nanostores/router to Reatom`?

**Comment by artalar** [comment]
@vostrik yep, take it! Write me DM for any questions.

**Comment by artalar** [comment]
@vostrik any news? =D

**Comment by vostrik** [comment]
Yo! I have already dive to the topic. I think that it will take a couple of weeks for the first demo.
